src/loop/typer.ml

```diff
@@ -73,6 +73,35 @@ module Zf_arith =
   Dolmen_type.Arith.Zf.Thf(T)
     (Dolmen.Std.Expr.Ty)(Dolmen.Std.Expr.Term)
 
+(* Extensions builtins *)
+(* ************************************************************************ *)
+
+module Ext = struct
+
+  type t = {
+    name : string;
+    builtins : Typer_intf.lang -> T.builtin_symbols;
+  }
+
+  let all = ref []
```

Comment:
Considering that one of the main use of extensions (in the cli) is loading them using their name, maybe a hashtable indexed by names would be better instead of a list here ? That would avoid surprising behaviour later if/when we start having a fair number of extensions (unlikely, but we never know).

Reply:
I simply reused the previous implementation here. I agree it would be better in the unlikely case we have a lot of extensions to use a hash table; I'd prefer changing the implementation in a follow-up PR if that's OK.

Reply:
that's ok

Reply:
Ended up actually doing it in this PR